filebeat/harvester/log.go

```diff
@@ -87,6 +88,11 @@ func (h *Harvester) Harvest() {
 		if !h.sendEvent(event) {
 			return
 		}
+
+		if h.config.CloseTimeout > 0 && time.Since(startTime) > h.config.CloseTimeout {
+			logp.Info("Closing harvester because ttl was reached: %s", h.path)
+			return
+		}
```

Comment:
Checking for CloseTimeout at this places requires an event to be published for closeTimeout to function.

If reader is waiting long for a new line or sendEvent blocks, due to spooler being blocked on publisher, close_timeout might be very very late.